### PR TITLE
Issue780 visualreport and invalid days

### LINKED ISSUE #780: visualreport and how it shows or suppresses invalid windows
The visualreport shows some invalid nights while also suppressing some nights, this seems to be specific when viewingwindow=2.

This seems especially confusing with recordings with a lot of missing day. I am currently investigating this.

### PR DESCRIPTION
**Pull request by vincentvanhees**
This PR fixes #780

1. I have added argument `visualreport_without_invalid` to either consistently show only valid nights/days (default) or to hide them.
2. I have tidied up the syntax of `g.plot5()` function, because it was really bad. In the future we may want to replace this function by an entirely new design for the visual report, but for the time being it seems good to make improvements like this PR in order to keep the code maintainable.
3. When the user tries to create the visual report for more than 50 files a message is shown to inform the them that this may take a while and that they may want to turn off the `visualreport`. Especially for larger datasets the visualreport feels to me as an annoying step of the process as people usually are not interested in the report when working with larger datasets and doing quantitative analysis. Hopefully this message will make them aware that the waiting time is not necessary.

Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Updated or expanded the documentation.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` and `CITATION.cff` files.

### CODE REVIEW

File: R/g.plot5.R [hunk 1]
```diff
@@ -91,60 +100,79 @@ g.plot5 = function(metadatadir = c(), dofirstpage = FALSE, viewingwindow = 1, f0
           }
           if (length(della) > 0) summarysleep_tmp = summarysleep_tmp[-della,]
         }
-        # do not include days with no meaningful data
         threshold_hrs_of_data_per_day = 0.5
```
Comment (jhmigueles) [inline R/g.plot5.R:103]: If I understand well, this requires half an hour with valid data for the day (full day, including day and spt), shouldn't this be consistent with `includedaycrit`?

File: R/g.plot5.R [hunk 2]
```diff
@@ -91,60 +100,79 @@ g.plot5 = function(metadatadir = c(), dofirstpage = FALSE, viewingwindow = 1, f0
...
-        if (length(tail_expansion_log) != 0) { # then keep timing of sleeponset to plot
-          lastnight = max(summarysleep_tmp$night)
-          if (lastnight %in% n2exclude) {
-            n2excludeb = n2exclude = n2exclude[-which(n2exclude == lastnight)]
+        if (visualreport_without_invalid == TRUE) {
+          incrementNight = 0
+          # do not include days with no meaningful data
+          d2excludeb = d2exclude = which(P2daysummary_tmp$N.valid.hours < threshold_hrs_of_data_per_day)
+          n2excludeb = n2exclude = which(summarysleep_tmp$fraction_night_invalid > 0.66
+                                         | summarysleep_tmp$SptDuration == 0)
```
Comment (jhmigueles) [inline R/g.plot5.R:109]: Should this 0.66 be `includenightcrit`?

**Review by jhmigueles — COMMENTED** [review]
Thanks for this work @vincentvanhees. I have added a couple of questions, other than that the script looks fine, and I have tested this in some files and it works as expected.

File: R/g.plot5.R (hunk 1 above)
Comment (vincentvanhees) [reply R/g.plot5.R:103]: Filtering based on `includedaycrit` happens further down in the code and is only done if `visualreport_without_invalid` is set to TRUE. However, the line here ensures that days are only plotted if they have at least half an hour of data regardless of whether they are valid or invalid. I remember inserting this as sometimes the first or last day of data only has one epoch of data.

File: R/g.plot5.R (hunk 2 above)
Comment (vincentvanhees) [reply R/g.plot5.R:112]: Good point, and I was also not using `includedaycrit`. I have now revised the code to use both and I have clarified the minimum per 0.5 hours inside the code as well.

**Review by jhmigueles — APPROVED** [review]
Thanks for the last changes! In the previous version I was not getting the invalid days removed because of the hard-coded definition of valid day/night, but now it is all working as expected.

**Pull request merged**